app/plugin.go

```diff
@@ -281,6 +282,36 @@ func (a *App) ShutDownPlugins() {
 	}
 }
 
+// ShutDownClientPlugins inform all clients that all plugins have been disabled.
+func (a *App) ShutDownClientPlugins() {
+	// Acquiring lock manually, as plugins might be disabled. See GetPluginsEnvironment.
+	a.Srv.PluginsLock.RLock()
+	pluginsEnvironment := a.Srv.PluginsEnvironment
+	a.Srv.PluginsLock.RUnlock()
+
+	if pluginsEnvironment == nil {
+		mlog.Debug("Plugins are already shutdown while trying to inform clients about this")
+		return
+	}
+
+	availablePlugins, err := pluginsEnvironment.Available()
+	if err != nil {
+		mlog.Error("Failed to get available plugins while trying to shut them down. Clients will not get notified about shut down plugins.", mlog.Err(err))
```

Comment:
Although it makes no difference for correctness, should we return early here for clarity?

Reply:
Sure :+1: